aws-athena/ctas/reporting-res_report_summary.sql

```diff
@@ -88,63 +47,91 @@ model_values AS (
+    WHERE NOT is_parking_space
+        AND NOT is_common_area
+),
+
+-- Join land, chars, and reporting groups to values
+all_values AS (
+    SELECT
+        fmvs.pin,
+        vptc.property_group,
+        vptc.class,
+        vptc.triad_name AS triad,
+        vptc.township_code,
+        CONCAT(vptc.township_code, vptc.nbhd) AS townnbhd,
+        fmvs.year,
+        fmvs.assessment_stage,
+        fmvs.total,
+        chars.yrblt,
+        chars.total_bldg_sf,
+        vpl.sf AS total_land_sf
+    FROM all_fmvs AS fmvs
+    LEFT JOIN {{ ref('reporting.vw_pin_township_class') }} AS vptc
+        ON fmvs.pin = vptc.pin
+        AND fmvs.year = vptc.year
+    INNER JOIN chars
+        ON fmvs.pin = chars.pin
+        AND fmvs.year = chars.year
+    LEFT JOIN {{ ref('default.vw_pin_land') }} AS vpl
+        ON fmvs.pin = vpl.pin
+        AND fmvs.year = vpl.year
+    WHERE vptc.property_group IS NOT NULL
+        AND vptc.triad_name IS NOT NULL
+),
+
+-- Sales, filtered to exclude outliers and mutlisales
+sales AS (
+    SELECT
+        vwps.sale_price,
+        vwps.year AS sale_year,
         tc.property_group,
-        aa.class,
-        tc.triad,
         tc.township_code,
-        tc.townnbhd,
-        aa.taxyr AS year,
-        CASE
-            WHEN aa.procname = 'CCAOVALUE' THEN 'mailed'
-            WHEN aa.procname = 'CCAOFINAL' THEN 'assessor certified'
-            WHEN aa.procname = 'BORVALUE' THEN 'bor certified'
-        END AS assessment_stage,
-        MAX(
-            CASE
-                WHEN aa.taxyr < '2020' THEN aa.ovrvalasm3
-                WHEN aa.taxyr >= '2020' THEN aa.valasm3
-            END
-        ) * 10 AS total
-    FROM {{ source('iasworld', 'asmt_all') }} AS aa
-    LEFT JOIN town_class AS tc
-        ON aa.parid = tc.parid
-        AND aa.taxyr = tc.taxyr
-    WHERE aa.procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
-        AND aa.rolltype != 'RR'
-        AND aa.deactivat IS NULL
-        AND aa.valclass IS NULL
+        vwps.nbhd AS townnbhd
+    FROM {{ ref('default.vw_pin_sale') }} AS vwps
+    LEFT JOIN {{ ref('reporting.vw_pin_township_class') }} AS tc
+        ON vwps.pin = tc.pin
+        AND vwps.year = tc.year
+    WHERE NOT vwps.is_multisale
+        AND NOT vwps.sale_filter_is_outlier
+        AND NOT vwps.sale_filter_deed_type
+        AND NOT vwps.sale_filter_less_than_10k
+        AND NOT vwps.sale_filter_same_sale_within_365
         AND tc.property_group IS NOT NULL
-        AND tc.triad IS NOT NULL
-        AND aa.taxyr >= '2021'
-    GROUP BY
-        aa.parid,
-        aa.taxyr,
-        aa.procname,
-        tc.property_group,
-        CASE
-            WHEN aa.procname = 'CCAOVALUE' THEN 'mailed'
-            WHEN aa.procname = 'CCAOFINAL' THEN 'assessor certified'
-            WHEN aa.procname = 'BORVALUE' THEN 'bor certified'
-        END,
-        aa.class,
-        tc.triad,
-        tc.township_code,
-        tc.townnbhd
+        AND tc.triad_name IS NOT NULL
```

Comment:
This is an artifact of `select count(*) from iasworld.legdat where user1 is null and cur = 'Y' and deactivat IS NULL` returning 404 rows. There are always `NULL` values for township and triad because of it they show up in aggregation which is frustrating.

Reply:
Very strange. It might be worth just manually updating those rows with the correct township code, to the extent that's possible. The we can use a dbt test to check for any new nulls.

Reply:
I corrected as many missing townships as possible for 2024. Test PINs and PINs with `99-999` for their neighborhoods will still have NULL `user1` fields, though, and I'm not sure the number of these rows will ever be static.